### PR TITLE
Set GitHub Status

### LINKED ISSUE #41: GitHub output for MVP
Option 1

Badge on the README. We just need a new endpoint on the site that checks the last job against a certain repo (we would need to store the github repo identifier somewhere)

Option 2 

Status API that shows valid/invalid and link to the job page. For this we need an authorized user to post statuses to the repo. See below

Authorization options

Option 2.1

We test repos on the frictionlessdata organization and use one of our hardcoded API keys to post Statuses

Option 2.2 

We implement OAuth Github authorization so users can authorize repo access and use their API tokens

### PR DESCRIPTION
**Pull request by amercader**
Fixes #41 

@roll I need to write some more tests and there is a failure caused by using dataset, but this is ready for review if you want to start having a look.

I've made some changes to the way job configs are prepared, what plugins can store in the DB and what the validation task returns. I recommend to follow the commits as I've separated

### CODE REVIEW

**Comment by amercader** [comment]
Tests fixed (with an ugly hack, I can't wait to get rid of dataset in #33)

**Comment by roll** [comment]
Ready for review? We could do #33 and #21 next. It's also useful to document our data model.

**Comment by amercader** [comment]
Ready for review, I'll add the tests later, I'm keen to deploy this to the staging site!

**Comment by roll** [comment]
OK. I'm reviewing.

Also we could add `goodtables.yml` to this repo and `data` folder with a few csv files to make it self-testable.

File: goodtablesio/helpers/create.py [hunk 1]
```diff
@@ -41,15 +46,18 @@ def create_job(validation_conf, job_id=None):
     return job_id
 
 
-def insert_job_row(job_id):
+def insert_job_row(job_id, plugin='api', plugin_conf=None):
     row = {
         'job_id': job_id,
+        'plugin': plugin,
```
Comment (roll) [inline goodtablesio/helpers/create.py]: plugin_name?

File: goodtablesio/helpers/prepare.py [hunk 2]
```diff
@@ -24,14 +24,27 @@ def prepare_job(job_conf, job_files):
...
+    base_url = job_conf_url.rsplit('/', 1)[0]
+    yml_file = _load_file(job_conf_url)
+    if yml_file:
+        job_conf = yaml.load(yml_file)
+
+        # Validate job configuration
+        validate_job_conf(job_conf)
+
+    else:
+        job_conf = {'files': '*'}
```
Comment (roll) [inline goodtablesio/helpers/prepare.py]: So it will fix #32

**Pull request merged**